samza-test/src/main/java/org/apache/samza/test/framework/StreamAssert.java

```diff
@@ -64,17 +70,30 @@
 
   private final String id;
   private final MessageStream<M> messageStream;
+  private final CollectionStream<M> collectionStream;
```

Comment:
The issue here is there is no common logic between assertions on MessageStreams an CollectionStream. To split them we have to provide something like CollectionStreamAssert and MessageStreamAssert classes

Reply:
Oh right, instead of "not work", I should have said "not ideal".
A couple of considerations for the user point of view:
1. If you want nice-looking test code: You could have a MessageStreamAssert class and a CollectionStreamAssert class, and then have the StreamAssert class have two "that" methods (one creates MessageStreamAssert, other creates CollectionStreamAssert). Since users just chain the methods, they won't actually see the terms "MessageStreamAssert" or "CollectionStreamAssert", but they will still use the methods from those classes.
2. Regarding confusion of having two classes: In my opinion, it is less confusing to have two classes, because then there is not really a question about which of the methods are usable for each use case, since the API forces users to use the correct ones.

Reply:
@shanthoosh pointed out that since this class StreamAssert does not maintain any state it will be better to just create static assertion utils, although this idea of StreamAssert being the builder for MessageStreamAssert and CollectionStreamAssert is great, we can use this in future once we have need for fluent assertions for ex

`StreamAssert.that(stream).function1().function1()..`